### PR TITLE
feat: charge transaction payment

### PR DESCRIPTION
**Pull request by failfmi**
**Detailed description**:
* Process fees when applying extrinsics

**Which issue(s) this PR fixes**:
Fixes #None

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: frame/balances/dispatchables/transfer.go [hunk 1]
```diff
@@ -227,8 +227,8 @@ func tryMutateAccount(who types.Address32, f func(who *types.AccountData, bool b
 
 	r := result.Value.(sc.VaryingData)
 
-	dustCleaner := r[1].(DustCleanerValue)
-	dustCleaner.Drop()
+	//dustCleaner := r[1].(DustCleanerValue)
```
Comment (radkomih) [inline frame/balances/dispatchables/transfer.go:232]: don't we need this?

File: runtime/runtime_test.go [hunk 2]
```diff
@@ -118,3 +120,18 @@ func setStorageAccountInfo(t *testing.T, storage *runtime.Storage, account []byt
 
 	return keyStorageAccount, accountInfo
 }
+
+func getQueryInfo(t *testing.T, runtime *wasmer.Instance, extrinsic []byte) primitives.RuntimeDispatchInfo {
+	buffer := &bytes.Buffer{}
+
+	buffer.Write(extrinsic)
+	sc.U32(buffer.Len()).Encode(buffer)
```
Comment (radkomih) [inline runtime/runtime_test.go:128]: Doesn't the extrinsic come with an already encoded length?

**Review by radkomih — DISMISSED** [review]

File: runtime/runtime_test.go (hunk 2 above)
Comment (failfmi) [reply runtime/runtime_test.go:128]: The API has a separate argument, specific for the length

File: frame/balances/dispatchables/transfer.go (hunk 1 above)
Comment (failfmi) [reply frame/balances/dispatchables/transfer.go:232]: Currently, it emits an event every time, which should not be the case. I'll add a TODO to fix it when the functionality is refactored.

**Review by radkomih — APPROVED** [review]

**Pull request merged**